config/deb.am

```diff
@@ -38,9 +43,14 @@ if CONFIG_USER
 	pkg7=$${name}-test-$${version}.$${arch}.rpm; \
 	pkg8=$${name}-dracut-$${version}.$${arch}.rpm; \
 	pkg9=$${name}-initramfs-$${version}.$${arch}.rpm; \
+	path_prepend=`mktemp -d /tmp/intercept.XXX`; \
+	ln -s /bin/true $${path_prepend}/dh_shlibdeps; \
```

Comment:
Let's make this discussion concrete. The packages which are detected as dependencies are (setting aside the internal dependencies of other zfs packages)

`liblkid1`, `libc6`, `libuuid1`, and `zlib1g`

These are all priority `required` Debian packages. So you are correct that the only concern is that they may be downgraded. This is unlikely---package managers will not do this automatically, they warn that it is is only to be performed by an "expert." **Do you have a reason to believe that a downgrade is more likely to cause a problem than an upgrade? Because the expert that is specifically telling their installation to downgrade may know better than you.**

> we're better off `make install`ing zfs direclty instead.

Although dependency tracking is one of the very nice features of package managers, it is not the only one. Debian advises against ever doing a `make install`, and instead recommends using [checkinstall](https://wiki.debian.org/CheckInstall) to produce a `deb`. No one is better off running that as root.

There are another few notable dependencies that are missing as well: **all of** the spl packages. Not to mention the **kernel modules** themselves. **You cannot possibly trust the dependencies** produced by `alien`---you're already forced to think about these yourself. Just putting the dependencies may lead a helpless newbie astray---and worse still put them in a position where they find their packages uninstallable, or pulling in the Debian versions automatically.

This pull request addresses a living concern for people who want to migrate from the Debian version of the packages to this upstream version, but who cannot purge the existing Debian packages from their machine.

> Then let's not make this a default for everyone until it's a better choice than the current one: can't we esclude just the conflicting libraries?

I'm guessing the decision to use `alien` was motivated out of frustration with the complexities of making a proper Debian package. There may be some way to pass excluded files through the `alien` call, into its `debian` directory, but that honestly buys us so little, and would cost so much time to investigate.